### PR TITLE
Alec - Added Edit Page for students

### LINKED ISSUE #49: SUBFEATURE: Edit page for Students plus tests and stories

### PR DESCRIPTION
**Pull request by alecsong222**
## Overview

This PR adds the Edit page frontend view to utilize the backend Edit functionality.
Branch - https://github.com/ucsb-cs156-f24/proj-happycows-f24-09/tree/Alec-StudentsEditPage

## Screenshots (Optional)

image

## Feedback Request (Optional)

Suggestions on how to handle the Course ID of a student when editing would be appreciated; since the Course ID of a student isn't allowed to be changed, I created a separate form for the edit page called StudentsEditForm.js and removed CourseID from there. If there are better approaches and/or things I could add please let me know.

## Validation (Optional)

Navigate to https://happycows-dev-alecsong222-students.dokku-09.cs.ucsb.edu and log in
Use Swagger to create a course, and a student corresponding with that course.
Go back to the home page, select Students from the Admin dropdown
You should be taken to the index page by default and see the student you just created
Edit the student you just created by clicking the Edit button in the table
Confirm that there is no "CourseID" field to be edited.
Submit your changes to your student and confirm that the changes have been saved.

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #49 

## PR Checklist
- [x] Commented code removed
- [x] PR deals with only one concern
- [x] PR Title is descriptive enough
- [x] PR description details what a user/dev will notice is different
- [x] No merge conflicts
- [x] Deployed and linked with Dokku deployment
- [x] New branch updated with main
- [x] PR is linked to a branch
- [x] PR is linked to an issue/ticket
- [x] Linked issue is in "In Review" column on Kanban board
- [x] PR is tagged with the Team's tag
- [x] All test cases pass

### CODE REVIEW

**Review by TomShangguan — DISMISSED** [review]
LGTM! I reviewed and checked the functionality of your code on dokku. It works fine.

**Review by elijahelephant — DISMISSED** [review]
As far as I can tell code doesn't have any glaring issues and dokku deployment works through my testing, approved and going to add ready for staff.

File: frontend/build-storybook.log [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+> frontend@0.1.0 build-storybook
+> build-storybook --docs -s public --output-dir=/var/folders/3b/x5mt4s9x3dx3ks913y481q340000gn/T/chromatic--65864-8YwseKd2A56g
+
```
Comment (pconrad) [inline frontend/build-storybook.log]: Remove from PR and consider adding to `.gitignore`

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove the irrelevant file, plus add this edit page into AppNavbar.js and App.js—otherwise, what's the point?

Also: we can't really test until it's integrated with the backend.

**Review by pconrad — DISMISSED** [review]
https://github.com/ucsb-cs156-f24/proj-happycows-f24-09/pull/43/files should not have been merged without adding validation for the courseId.

Something like this:
```
        coursesRepository.findById(courseId).orElseThrow(() -> new EntityNotFoundException(Courses.class, courseId));
```

I won't revoke the points you already earned.  But I also won't merge anything else that touches CRUD operations for Student (front end or backend) unless until this fix is part of that PR.

File: frontend/src/main/components/Students/StudentsEditForm.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        { defaultValues: initialContents || {}, }
+    );
+    // Stryker restore all
+
+    const navigate = useNavigate();
+
+    const testIdPrefix = "StudentsEditForm";
+
+    // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+    // Note that even this complex regex may still need some tweaks
```
Comment (pconrad) [inline frontend/src/main/components/Students/StudentsEditForm.js]: These comments appear to be leftover from code that was deleted.  Can we remove these?

**Review by pconrad — DISMISSED** [review]
One very small change...

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**